src/librustc_resolve/lib.rs

```diff
@@ -2167,18 +2167,20 @@ impl<'a> Resolver<'a> {
                     let self_is_available = this.self_value_is_available(path[0].ctxt);
                     match candidate {
                         AssocSuggestion::Field => {
-                            err.span_label(span, &format!("did you mean `self.{}`?", path_str));
+                            err.guess(span, "did you intend to access a struct field?",
+                                      format!("self.{}", path_str));
                             if !self_is_available {
                                 err.span_label(span, &format!("`self` value is only available in \
                                                                methods with `self` parameter"));
                             }
                         }
                         AssocSuggestion::MethodWithSelf if self_is_available => {
-                            err.span_label(span, &format!("did you mean `self.{}(...)`?",
-                                                           path_str));
+                            err.guess(span, "did you intend to call a method of the same name?",
+                                      format!("self.{}", path_str));
```

Comment:
I though that the span was for the entire call, instead of just the `self.{}` part